sns/parity_math.c

```diff
@@ -405,9 +832,18 @@ static void reed_solomon_diff(struct m0_parity_math *math,
 	M0_PRE(new    != NULL);
 	M0_PRE(parity != NULL);
 	M0_PRE(index  <  math->pmi_data_count);
-	M0_PRE(old[index].b_nob == new[index].b_nob);
-	M0_PRE(m0_forall(i, math->pmi_parity_count,
-		         new[index].b_nob == parity[i].b_nob));
+
+	if (old[index].b_nob != new[index].b_nob)
+		return M0_ERR_INFO(-EINVAL, "Data block size mismatch. "
+				   "Index=%u, Old data block size=%u, "
+				   "New data block size=%u", index,
+				   (uint32_t)old[index].b_nob,
+				   (uint32_t)new[index].b_nob);
+
+	for (pi = 0; pi < math->pmi_parity_count; ++pi)
+		if (new[index].b_nob != parity[pi].b_nob)
+			return BLOCK_SIZE_ERR_INFO("parity", new[index].b_nob, pi,
+						   parity[pi].b_nob);
```

Comment:
Why? Original assertions are correct.

Reply:
This change was done as part of PR #741.

Reply:
Updated the code to assert with additional info.